src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1625,7 +1625,7 @@ static bool IsDeclaredWithinType (IMemberDefinition member, TypeDefinition type)
 			var memberScope = reportOnMember ? ScopeStack.PushScope (new MessageOrigin (member)) : null;
 
 			try {
-				var origin = ScopeStack.CurrentScope.Origin;
+				origin = reportOnMember ? ScopeStack.CurrentScope.Origin : origin;
```

Comment:
Nit: I think it might be better to just leave: `origin = ScopeStack.CurrentScope.Origin;`

At the beginning of the method, `origin` is equal to `ScopeStack.CurrentScope.Origin`, then we attempt to push into the stack depending on `reportOnMember`. Either way `ScopeStack.CurrentScope.Origin` contains the right value, either contains origin or contains the new MessageOrigin.

Reply:
`origin` might not be equal to `ScopeStack.CurrentScope.Origin` if the caller didn't push the scope to the scope stack - so I only wanted this to look at the scope stack if we know we pushed to it.

Reply:
While looking at it I realized this scope stack usage was no longer necessary, so I simplified the logic some. Thanks!

Reply:
What I'm saying is that is safe to always assign `origin` to `ScopeStack.CurrentScope.Origin` because that value is either `origin` or `new MessageOrigin (member)`

Reply:
New changes look good to me, thanks! :D